tronweb/coffee/graph.coffee

```diff
@@ -1,126 +1,324 @@
+        # supports setting the node width as 'label'. See https://github.com/cytoscape/cytoscape.js/issues/2713
+        ctx = document.createElement('canvas').getContext("2d")
+        fStyle = node.pstyle('font-style').strValue
+        size = node.pstyle('font-size').pfValue + 'px'
+        family = node.pstyle('font-family').strValue
+        weight = node.pstyle('font-weight').strValue
+        ctx.font = "#{fStyle} #{weight} #{size} #{family}"
 
-    el: "#action-graph"
+        return ctx.measureText(node.data('name')).width + 20
+
+    # Search based on node name. Highlights matching nodes and their connected edges, dims the rest.
+    applySearch: (cy, searchText) ->
+        if searchText
+            cy.nodes().forEach (node) ->
+                nodeName = node.data('name').toLowerCase()
+                if nodeName.includes(searchText)
+                    node.style('opacity', 1)
+                else
+                    node.style('opacity', 0.2)
+
+            cy.edges().style('opacity', 0.1)
+
+            matchingNodes = cy.nodes().filter (node) ->
+                node.data('name').toLowerCase().includes(searchText)
+
+            matchingNodes.connectedEdges().style('opacity', 0.8)
+        else
+            # Reset all styles when search is cleared
+            cy.nodes().style('opacity', 1)
+            cy.edges().style('opacity', 1)
+
+    # Reset the graph to its initial state, including layout and node positions
+    resetGraph: (cy) ->
+        cy.stop()
+        cy.nodes().style('opacity', 1)
+        cy.edges().style('opacity', 1)
+
+        # Reset node positions before re-layout
+        cy.nodes().positions (node) ->
+            return { x: 0, y: 0 }
+
+        layout = cy.layout(module.GraphUtils.getDefaultLayout())
+        layout.run()
+
+        # Fit all elements (if no elements are specified it fits all, so we pass undefined) with 25px padding
+        cy.fit(undefined, 25)
+        cy.center()
+
+        cy.nodes().forEach (node) ->
+            node.data 'manuallyPositioned', false
+
+    defaultStylesheet: [
+        {
+            selector: 'node',
+            style: {
+                'label': 'data(name)'
+                'text-valign': 'center'
+                'text-halign': 'center'
+                'background-color': '#f8f8f8'
+                'color': '#000'
+                'font-size': '16px'
+                'font-weight': 'bold'
+                'shape': 'roundrectangle'
+                'width': (node) -> module.GraphUtils.getNodeWidth(node)
+                'height': 25
+                'padding-top': '5px'
+                'padding-bottom': '5px'
+                'padding-left': '10px'
+                'padding-right': '10px'
+                'text-wrap': 'none'
+                'border-width': 3
+                'border-color': '#999'
+            }
+        },
+        {
+            selector: 'edge',
+            style: {
+                'curve-style': 'bezier'
```

Comment:
Hey wait a minute, why aren't you working?

Reply:
Context, I was expecting this to curve lines around nodes a bit based on cytoscape's [demo](https://js.cytoscape.org/demos/edge-types/), but I think I've misunderstood things here.

At any rate, it's probably more advisable to go with taxi edges which looks like [this](https://fluffy.yelpcorp.com/i/8RwgfbkgrTF081qDgT5xHGVWfbxtlvSm.html), or something like an unbundled bezier with dynamic curves which would look [this](https://user-images.githubusercontent.com/989043/73289286-77f58600-41ca-11ea-960c-563194cda7f2.png).